### PR TITLE
chore: Upgrade core packages to release 159.0.0

### LINKED ISSUE #25401: Upgrade core packages to release 159.0.0
### What is this about?

Some time ago, the Wallet Framework team released version 159.0.0 of the `core` repo, which bumped all `core` packages to use a minimum version of Node 18. We need to ensure that among the packages that the extension are using, the versions of these packages match at least those that were included in release 159.0.0.

### Scenario

_No response_

### Design

_No response_

### Technical Details

We do not need to upgrade the extension to Node 18, as that is already the case.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

These packages are upgraded:

- [ ] `@metamask/address-book-controller` (4.0.2 -> 5.0.0)
- [ ] `@metamask/announcement-controller` (6.1.1 -> 7.0.0)
- [ ] `@metamask/assets-controller` (30.0.0 -> 32.0.0)
- [ ] `@metamask/base-controller` (5.0.2 -> 6.0.0)
- [ ] `@metamask/build-utils` (2.0.1 -> 3.0.0)
- [ ] `@metamask/controller-utils` (10.0.0 -> 11.0.0)
- [ ] `@metamask/ens-controller` (11.0.0 -> 12.0.0)
- [ ] `@metamask/eth-json-rpc-provider` (3.0.2 -> 4.0.0)
- [ ] `@metamask/gas-fee-controller` (16.0.0 -> 17.0.0)
- [ ] `@metamask/json-rpc-engine` (8.0.2 -> 9.0.0)
- [ ] `@metamask/json-rpc-middleware-stream` (7.0.2 -> 8.0.0)
- [ ] `@metamask/keyring-controller` (16.1.0 -> 17.0.0)
- [ ] `@metamask/logging-controller` (4.0.0 -> 5.0.0)
- [ ] `@metamask/message-manager` (9.0.0 -> 10.0.0)
- [ ] `@metamask/name-controller` (7.0.0 -> 8.0.0)
- [ ] `@metamask/notification-controller` (5.0.2 -> 6.0.0)
- [ ] `@metamask/phishing-controller` (9.0.4 -> 10.0.0)
- [ ] `@metamask/preferences-controller` (12.0.0 -> 13.0.0)
- [ ] `@metamask/queued-request-controller` (0.11.0 -> 0.12.0)
- [ ] `@metamask/rate-limit-controller` (5.0.2 -> 6.0.0)
- [ ] `@metamask/signature-controller` (17.0.0 -> 18.0.0)
- [ ] `@metamask/transaction-controller` (31.0.0 -> 32.0.0)
- [ ] `@metamask/user-operation-controller` (11.0.0 -> 12.0.0)

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by cryptodev-2s**
Upgrade core packages to release 159.0.0.

Upgraded packages:

-  @metamask/address-book-controller (4.0.2 -> 5.0.0)
-  @metamask/announcement-controller (6.1.0-> 7.0.0)
-  @metamask/base-controller (5.0.1 -> 6.0.2)
-  @metamask/message-manager (7.3.0 -> 10.0.0)
-  @metamask/notification-controller (5.0.1 -> 6.0.0)
-  @metamask/rate-limit-controller (5.0.1 -> 6.0.0)
- @metamask/phishing-warning: (3.0.3 -> 4.0.0),
-  @metamask/build-utils (1.0.0 -> 3.0.0)

## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26138?quickstart=1)

## **Related issues**

Fixes: [#25401](https://github.com/MetaMask/metamask-extension/issues/25401)

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

File: .yarn/patches/@metamask-controller-utils-npm-8.0.4-077edd6162.patch
Comment (Gudahtt) [inline .yarn/patches/@metamask-controller-utils-npm-8.0.4-077edd6162.patch:1]: Strange, it looks like both of these patches were unused? I guess they were left behind from previous updates, good catch

File: package.json [hunk 1]
```diff
@@ -287,15 +287,15 @@
...
+    "@metamask/announcement-controller": "^7.0.0",
     "@metamask/approval-controller": "^7.0.0",
     "@metamask/assets-controllers": "^36.0.0",
-    "@metamask/base-controller": "^5.0.1",
+    "@metamask/base-controller": "^6.0.2",
     "@metamask/bitcoin-wallet-snap": "^0.3.0",
     "@metamask/browser-passworder": "^4.3.0",
     "@metamask/contract-metadata": "^2.5.0",
-    "@metamask/controller-utils": "^10.0.0",
+    "@metamask/controller-utils": "^11.0.2",
```
Comment (Gudahtt) [inline package.json:298]: A number of these packages have breaking changes related to the removal of `eth_sign`. It seems that we may be blocked by this PR: https://github.com/MetaMask/metamask-extension/pull/24756

Comment (cryptodev-2s) [reply package.json:298]: Yes indeed it's the case I wasn't able to upgrade `signature-controller` for the same reason

Comment (Gudahtt) [reply package.json:298]: Got it, OK, well at least that PR seems close to being merged

**Comment by MajorLift** [comment]
Migrating `json-rpc-engine` to `@metamask/json-rpc-engine` has been previously attempted,
- https://github.com/MetaMask/metamask-extension/pull/22875
- https://github.com/MetaMask/metamask-extension/pull/23532

but postponed due to the possibility of breaking dapps:
- https://consensys.slack.com/archives/GN3SR3GNM/p1710780668884679
- https://consensys.slack.com/archives/C01V1L10W2E/p1719347208154459

Has this issue been resolved?

I believe [`eth-rpc-errors` to `@metamask/rpc-errors` migration is a blocker for this](https://consensys.slack.com/archives/C01V1L10W2E/p1719424100562229?thread_ts=1719347208.154459&cid=C01V1L10W2E), and that PR hasn't been merged yet.
- https://github.com/MetaMask/metamask-extension/pull/24496

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -4996,11 +4996,11 @@ export default class MetamaskController extends EventEmitter {
     if (sender.url) {
       if (this.onboardingController.store.getState().completedOnboarding) {
         if (this.preferencesController.store.getState().usePhishDetect) {
-          const { hostname } = new URL(sender.url);
           this.phishingController.maybeUpdateState();
           // Check if new connection is blocked if phishing detection is on
-          const phishingTestResponse = this.phishingController.test(hostname);
+          const phishingTestResponse = this.phishingController.test(sender.url);
```
Comment (cryptodev-2s) [inline app/scripts/metamask-controller.js:5001]: @Gudahtt, the `test` method in `phishingDetector` was updated to accept a URL in this [PR](https://github.com/MetaMask/core/pull/4465) [see change](https://github.com/MetaMask/core/pull/4465/files#diff-a3e0475a00b3bf152d951e242c271df7621313a1cbc20673c1bb882f935bd740R147). However, I believe this is a breaking change that should have been mentioned in the README.

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

File: app/scripts/lib/ppom/ppom-util.test.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import {
   normalizeTransactionParams,
 } from '@metamask/transaction-controller';
 import { SignatureController } from '@metamask/signature-controller';
-import { Message } from '@metamask/message-manager';
+import { PersonalMessage } from '@metamask/message-manager';
```
Comment (MajorLift) [inline app/scripts/lib/ppom/ppom-util.test.ts:10]: nit:
```suggestion
import type { PersonalMessage } from '@metamask/message-manager';
```

File: app/scripts/lib/tx-verification/tx-verification-middleware.ts [hunk 4]
```diff
@@ -3,17 +3,17 @@ import { verifyMessage } from '@ethersproject/wallet';
...
 } from '@metamask/utils';
 import {
   JsonRpcRequest,
-  JsonRpcResponse,
-  JsonRpcEngineEndCallback,
-  JsonRpcEngineNextCallback,
+  type JsonRpcResponse,
+  type JsonRpcEngineEndCallback,
+  type JsonRpcEngineNextCallback,
 } from 'json-rpc-engine';
```
Comment (MajorLift) [inline app/scripts/lib/tx-verification/tx-verification-middleware.ts:17]: nit:
```suggestion
import type {
  Json,
  JsonRpcParams,
  Hex,
} from '@metamask/utils';
import {
  hasProperty,
  isObject,
} from '@metamask/utils';
import type {
  JsonRpcResponse,
  JsonRpcEngineEndCallback,
  JsonRpcEngineNextCallback,
} from 'json-rpc-engine';
import {
  JsonRpcRequest,
} from 'json-rpc-engine';
```

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

File: app/scripts/lib/snap-keyring/utils/isBlockedUrl.test.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ describe('isBlockedUrl', () => {
     allowedEvents: [],
   });
   const phishingController = new PhishingController({
-    // @ts-expect-error TODO: Resolve/patch mismatch between messenger types
+    // @ts-expect-error TODO: Resolve mismatch between base-controller versions.
```
Comment (MajorLift) [inline app/scripts/lib/snap-keyring/utils/isBlockedUrl.test.ts:13]: Note: `@metamask/phishing-controller` is now on `@metamask/base-controller@7.0.0`, while the rest of extension is using `@metamask/base-controller@6.0.2`.

**Review by MajorLift — APPROVED** [review]
LGTM!

File: app/scripts/lib/snap-keyring/utils/isBlockedUrl.test.ts (hunk 5 above)
Comment (cryptodev-2s) [reply app/scripts/lib/snap-keyring/utils/isBlockedUrl.test.ts:13]: Yes exactly

**Review by Mrtenz — APPROVED** [review]

**Review by montelaidev — APPROVED** [review]
Accounts changes LGTM

**Pull request merged**